File: frontend/src/shared/domain/group.ts
// ============================================================
// group.ts — domain entity cho Group và GroupMember
// ============================================================

import type { User } from "@features/user";
import type { GroupRole } from "@shared/domain/enums";
import type { ID, Timestamp } from "@shared/utils/common";

export interface Group {
    id: ID;
    classId: ID;
    name: string | null;
    leader: User;
    createdAt: Timestamp;
}

export interface GroupMember {
    id: ID;
    user: User;
    role: GroupRole;
    joinedAt: Timestamp;
}

File: backend/src/main/java/com/mezon/classmanagement/backend/domain/group/entity/Group.java
package com.mezon.classmanagement.backend.domain.group.entity;

import com.mezon.classmanagement.backend.domain.clazz.entity.Class;
import com.mezon.classmanagement.backend.domain.auth.entity.User;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.FieldDefaults;

import java.time.Instant;

@Entity
@FieldDefaults(level = AccessLevel.PRIVATE)
@Setter
@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "groups")
public class Group {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "id", nullable = false)
	Long id;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "class_id", nullable = false)
	Class clazz;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "leader_user_id", nullable = false)
	User leader;

	@Column(name = "name", nullable = true)
	String name;

	@Column(name = "created_at", nullable = false, insertable = false, updatable = false)
	Instant createdAt;

}